torch/_dynamo/variables/builtin.py

```diff
@@ -473,6 +473,9 @@ def call_function(
                     # Work around weird bug in hf_T5
                     fn, args = operator.add, [args[1], args[0]]
 
+                if self.fn is operator.not_:
+                    fn = torch.logical_not
```

Comment:
Is `logical_not` a good replacement for  `operator.not_`? `operator.not_` works only on 1-element tensors, `logical_not` will happily work on any size tensors, and hopefully it would error out downstream of this callsite, but are we sure about that?

Reply:
Is it ok to assume that the programs users pass into export are valid programs? If we assume that, then we would know that the tensor passed in is 1-element since it passes an eager run with `operator.not_`. If not then could we do an eager run before doing the actual exporting to ensure that the program is passing in valid values to the `not`?